libxdk/include/xdk/util/pwn_utils.h

```diff
@@ -48,4 +48,13 @@ uint64_t check_kaslr_base(uint64_t kbase_addr);
  * @param heap_leak The address to check.
  * @return The checked kernel heap pointer if valid.
  */
-uint64_t check_heap_ptr(uint64_t heap_leak);
\ No newline at end of file
+uint64_t check_heap_ptr(uint64_t heap_leak);
+
+/**
+ * @ingroup util_classes
+ * @brief Pins the calling thread to the provided CPU.
+ *
+ * @param cpu The CPU the calling thread should be pinned to. Defaults to 0 if unspecified.
+ * @throws errno_error if the operation fails
+ */
+void pin_cpu(int cpu = 0);
```

Comment:
I would remove defaulting to zero, because pinning to CPU must be a very specific (you typically pin one thread to 0, and the other to 1), and with a default argument it's less clear what's going on.

Reply:
Makes sense, done.